h2o-genmodel/src/main/java/hex/genmodel/easy/EasyPredictModelWrapper.java

```diff
@@ -645,7 +645,7 @@ private void validateModelCategory(ModelCategory c) throws PredictException {
           else {
             value = levelIndex;
           }
-        } else if (o instanceof Double && Double.isNaN((double)o)) {
+        } else if (o instanceof Double && !Double.isNaN((double)o)) {
```

Comment:
May you please explain your motivation and attach a test-case for your situation? So far the change does not make sense.

Reply:
Agreed with @mmalohlava - this change is not correct in my opinion.

This branch of execution handles _categorical_ variables. If an instance of Double gets in there - it _has to be NaN_. Double.NaN is our representation of a missing value.